### PR TITLE
more micro grabbag fixes

### PR DESCRIPTION
**Pull request by vtjnash**

### CODE REVIEW

**Comment by Keno** [comment]
a805aa6 probably needs some discussion. The others LGTM.

**Comment by vtjnash** [comment]
@Keno I can pull it out into a separate PR, but I'm not really expecting to get any significant feedback on that commit.

**Comment by Keno** [comment]
I took a quick at that commit as well and I'm generally fine with it. I do think @JeffBezanson should have a look also though.

**Comment by JeffBezanson** [comment]
+1

My only reaction is that it seems odd to move Timer to locks.jl. A timer is not a lock. I can kind of see why AsyncCondition might be in there, but it's also not a lock. So +1 for the rename and removal of callback API, -1 to the code movement.

**Comment by vtjnash** [comment]
My main goal was to move it out of stream.jl, but wasn't sure if it should land in tasks.jl or stay with AsyncCondition and be part of Threads. Perhaps locks.jl should be renamed synchronize.jl?

**Comment by JeffBezanson** [comment]
Maybe an events.jl? I don't think it makes sense to group locks and timers.

File: deps/Makefile [hunk 1]
```diff
@@ -32,6 +32,11 @@ include $(SRCDIR)/llvm-ver.make
 
 ## Some shared configuration options ##
 
+# don't pass makefile flags through by default
+# (which would often result in unintentionally overriding the flags we are trying to set here)
+MAKEOVERRIDES =
```
Comment (tkelman) [inline deps/Makefile]: I still think this violates distro policies

Comment (vtjnash) [inline deps/Makefile]: most distros don't want to build the contents of deps.

this is more predictable than the current situation (which can lead to nasty surprises), although i entirely expect that more work will need to happen over time to address other cases as they come up (as mentioned in the commit)

Comment (tkelman) [inline deps/Makefile]: > most distros don't want to build the contents of deps.

except bundled things like libuv where we still can't use upstream versions

Comment (tkelman) [inline deps/Makefile]: we have a `DEPS_CFLAGS`, we should try actually using it

Comment (vtjnash) [inline deps/Makefile]: putting flags into the `CC` variable isn't terribly portable (aka `cmake` chokes on it). we should probably just remove that.

Comment (nalimilan) [inline deps/Makefile]: I think I would also prefer we pass CFLAGS manually to dependencies. That way, the worse that can happen is a build failure, which doesn't go unnoticed. OTOH, when setting `MAKEOVERRIDES`, dependencies will silently not use the distribution CFLAGS, which can give more obscure issues (like ABI discrepancies on ARM, missing security-related flags, etc.).

Anyway most distributions only need to build libuv, and maybe libunwind until upstream makes a new release, so we can just fix these two cases for now.

**Comment by vtjnash** [comment]
@JeffBezanson the tests are failing on this PR because the `@async` block is creating a new variable instead of capturing the intended one from the let block (due to the presence of a global of the same name):

```julia
julia> t = 3
3

julia> let t
           c = Condition()
           task = @async begin
               notify(c)
               t = Timer(0.055)
               wait(t)
           end
           wait(task)
           println(t)
       end
ERROR: UndefVarError: t not defined
 in eval(::Module, ::Any) at ./boot.jl:243
```

**Comment by vtjnash** [comment]
bump. ready to merge.

**Comment by nalimilan** [comment]
Don't you want to remove the Makefile change until we find the best solution?

File: deps/Makefile (hunk 1 above)
Comment (vtjnash) [inline deps/Makefile]: makes sense – we were already forwarding CFLAGS to libuv (twice in fact, to try to avoid this breakage condition)

**Comment by vtjnash** [comment]
oh right, i forgot i hadn't responded above. i don't think there can be a "best" solution, but i think the best approach is to merge this, then work on addressing future issues as them come up.

**Comment by tkelman** [comment]
Immediate future issue: deps/Makefile does not respect flags from environment.

**Comment by vtjnash** [comment]
Immediate current issue: flags from the environment break some of deps/Makefile

**Comment by tkelman** [comment]
Please take that out of this PR, since it's completely unrelated to the other 3 commits which seem less controversial. We can find a less breaking fix for that issue on its own. I'm concerned that this may break the from-scratch build on Travis, but testing that would require clearing out the cache.

**Pull request closed without merging**